tests/integration/feeds/test_feeds_client.py

```diff
@@ -0,0 +1,270 @@
+            platform=platform,
+            description=description,
+        )
+        return feed_request
+
+    yield _create_feed_request
+
+
+@pytest.fixture(scope="class")
+def binary_pkg_file_data() -> BinaryIO:
+    """Fixture to return package file in binary format."""
+    package_data = open(PACKAGE_PATH, "rb")
+    return package_data
+
+
+@pytest.fixture(scope="class")
+def invalid_id() -> str:
+    """Generate a invalid id."""
+    id = f"Invalid-id-{randint(1000, 9999)}"
+    return id
+
+
+@pytest.fixture(scope="class")
+def get_feed_name():
+    """Generate a feed name."""
+    name = "Test_Feed_"
+    feed_count = 0
+
+    def _get_feed_name():
+        nonlocal feed_count
+        feed_count += 1
+        feed_name = name + str(feed_count)
+        return feed_name
+
+    yield _get_feed_name
+
+
+@pytest.mark.enterprise
+@pytest.mark.integration
+class TestFeedsClient:
+    """A set of test methods to test SystemLink Feeds API."""
+
+    def test__create_feed_windows_platform__succeeds(
+        self,
+        create_feed: Callable,
+        create_feed_request: Callable,
+        get_feed_name: Callable,
+    ):
+        """Test the case of a completely successful create feed API for Windows platform."""
+        request_body = create_feed_request(
+            feed_name=get_feed_name(),
+            description=FEED_DESCRIPTION,
+            platform=Platform.WINDOWS,
+        )
+        response = create_feed(request_body)
+
+        assert response.id is not None
+        assert response.workspace is not None
+        assert response.name is not None
+        assert response.platform == Platform.WINDOWS
+        assert response.description == FEED_DESCRIPTION
+
+    def test__create_feed_linux_platform__succeeds(
+        self,
+        create_feed: Callable,
+        create_feed_request: Callable,
+        get_feed_name: Callable,
+    ):
+        """Test the case of a completely successful create feed API for Linux platform."""
+        request_body = create_feed_request(
+            feed_name=get_feed_name(),
+            description=FEED_DESCRIPTION,
+            platform=Platform.NI_LINUX_RT,
+        )
+        response = create_feed(request_body)
+
+        assert response.id is not None
+        assert response.workspace is not None
+        assert response.name is not None
+        assert response.platform == Platform.NI_LINUX_RT
+        assert response.description == FEED_DESCRIPTION
+
+    def test__query_feeds_windows_platform__succeeds(
+        self,
+        client: FeedsClient,
+        create_feed: Callable,
+        create_feed_request: Callable,
+        get_feed_name: Callable,
+    ):
+        """Test the case for querying available feeds for Windows platform."""
+        create_feed_request_body = create_feed_request(
+            feed_name=get_feed_name(),
+            description=FEED_DESCRIPTION,
+            platform=Platform.WINDOWS,
+        )
+        create_feed(create_feed_request_body)
```

Comment:
do a sanity check for `id` to ensure the feed was created before checking `query` works.